Assets/Scripts/Battlefield/CreatureScripts/UniqueCreature.cs

```diff
@@ -55,6 +56,14 @@ public void Glow(int glow)
 
         private void Update()
         {
+            if (health <= 0)
+            {
+                Destroy(target.gameObject);
+                Destroy(currentCamera.gameObject);
+                Destroy(transform.gameObject);
+            }
+
+
```

Comment:
It is safer to do this when the Creature takes damage and not every single frame. Also, we could have special effect that  activate when health drops sometimes therefore I suggest this be much more controlled and testing only when damage is dealt.